src/Drivers/OpenAI/OpenAI.php

```diff
@@ -68,11 +68,14 @@ public function text(TextRequest $request): DriverResponse
             ));
         }
 
-        $toolCalls = array_map(fn (array $toolCall): ToolCall => new ToolCall(
-            id: data_get($toolCall, 'id'),
-            name: data_get($toolCall, 'function.name'),
-            arguments: data_get($toolCall, 'function.arguments'),
-        ), data_get($data, 'choices.0.message.tool_calls', []));
+        $toolCalls = [];
+        if ($rawToolCalls = data_get($data, 'choices.0.message.tool_calls')) {
+            $toolCalls = array_map(fn (array $toolCall): ToolCall => new ToolCall(
+                id: data_get($toolCall, 'id'),
+                name: data_get($toolCall, 'function.name'),
+                arguments: data_get($toolCall, 'function.arguments'),
+            ), $rawToolCalls);
+        }
```

Comment:
@pushpak1300 can you explain the reasoning behind this change?

Reply:
@sixlive in case if tool call is not there Mistral API send empty array as a tool property.

Reply:
I think we could do this instead, yeah?

```php
        $toolCalls = array_map(fn (array $toolCall): ToolCall => new ToolCall(
            id: data_get($toolCall, 'id'),
            name: data_get($toolCall, 'function.name'),
            arguments: data_get($toolCall, 'function.arguments'),
        ), data_get($data, 'choices.0.message.tool_calls', []) ?? []);
```

Reply:
Pushed the change @sixlive